lib/src/clojure_lsp/kondo.clj

```diff
@@ -130,29 +138,29 @@
       (assoc-in [:config :linters :unresolved-namespace :report-duplicates] true)
       (assoc-in [:config :linters :unresolved-var :report-duplicates] true))))
 
+(defn ^:private run-custom-lint? [config]
+  (not= :off (get-in config [:linters :clojure-lsp/unused-public-var :level])))
+
 (defn ^:private custom-lint-project!
-  [external-analysis-only? {:keys [analysis config] :as kondo-ctx}]
-  (when-not (= :off (get-in config [:linters :clojure-lsp/unused-public-var :level]))
-    (let [new-analysis (group-by :filename (normalize-analysis external-analysis-only? analysis))]
-      (f.diagnostic/custom-lint-project! new-analysis kondo-ctx))))
+  [{:keys [config analysis] :as kondo-ctx}]
+  (when (run-custom-lint? config)
+    (shared/logging-time
+      "Linting whole project for unused-public-var took %s"
+      (f.diagnostic/custom-lint-project! analysis kondo-ctx))))
 
 (defn ^:private custom-lint-files!
-  [files db* {:keys [analysis] :as kondo-ctx}]
-  (shared/logging-task
-    :reference-files/lint
-    (let [db @db*
-          new-analysis (group-by :filename (normalize-analysis false analysis))
-          updated-analysis (merge (:analysis db) new-analysis)]
-      (f.diagnostic/custom-lint-files! files updated-analysis kondo-ctx))))
+  [files db {:keys [config analysis] :as kondo-ctx}]
+  (when (run-custom-lint? config)
```

Comment:
This didn't check whether to run custom lint before. I think this is a bug fix?

Reply:
Yes, I probably missed that when refactoring references lint feature 😅